oxcaml opam ci reports conflicts

This has begun to occur recently in every PR:

```

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[dune.3.23.0] synchronised (no changes)

[ERROR] Package conflict!
  * Incompatible packages:
    - (invariant) → ocaml-variants = 5.2.0+ox → oxcaml-compiler → oxcaml-patch-guards → oxcaml-alcotest → oxcaml-backoff → oxcaml-chrome-trace → oxcaml-cmarkit → oxcaml-ctypes → oxcaml-ctypes-foreign → oxcaml-dot-merlin-reader → oxcaml-dune
    - (invariant) → ocaml-variants = 5.2.0+ox → oxcaml-compiler → oxcaml-patch-guards → oxcaml-alcotest-patches → alcotest = 1.9.0+ox → dune >= 3.0
    You can temporarily relax the switch invariant with `--update-invariant'
  * Missing dependency:
    - (invariant) → ocaml-variants = 5.2.0+ox → oxcaml-compiler → oxcaml-patch-guards → oxcaml-alcotest → oxcaml-backoff → oxcaml-chrome-trace-patches → chrome-trace = 3.21.0+ox
    not available because the package is pinned to version 3.23.0
  * Missing dependency:
    - (invariant) → ocaml-variants = 5.2.0+ox → oxcaml-compiler → oxcaml-patch-guards → oxcaml-alcotest → oxcaml-backoff → oxcaml-chrome-trace → oxcaml-cmarkit → oxcaml-ctypes → oxcaml-ctypes-foreign → oxcaml-dot-merlin-reader → oxcaml-dune-patches → dune (>= 3.20.2+ox & != 3.21.0 & < 3.21.1)
    not available because the package is pinned to version 3.23.0

```

We need to investigate why and fix it. See here for an example https://github.com/ocaml/dune/actions/runs/24691695530/job/72215292560?pr=14248

I've provided an alternative oxcaml job for the time being here:
- https://github.com/ocaml/dune/pull/14248

We could consider placing it entirely.

--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
@@ -425,11 +425,8 @@ jobs:
         with:
           node-version: latest
 
-      - name: Pin deps
-        run: opam pin add -n . --with-version=3.23.0
-
       - name: Install deps
-        run: opam install csexp pp re spawn uutf ppx_expect ppx_inline_test dune wasm_of_ocaml-compiler
+        run: opam install csexp pp re spawn uutf ppx_expect ppx_inline_test wasm_of_ocaml-compiler
 
       - name: Build dune
         run: opam exec -- make bootstrap

--- a/.github/workflows/workflow.yml.in
+++ b/.github/workflows/workflow.yml.in
@@ -424,11 +424,8 @@ jobs:
         with:
           node-version: latest
 
-      - name: Pin deps
-        run: opam pin add -n . --with-version=%%VERSION%%
-
       - name: Install deps
-        run: opam install csexp pp re spawn uutf ppx_expect ppx_inline_test dune wasm_of_ocaml-compiler
+        run: opam install csexp pp re spawn uutf ppx_expect ppx_inline_test wasm_of_ocaml-compiler
 
       - name: Build dune
         run: opam exec -- make bootstrap

